src/loongarch64/Gresume.c

```diff
@@ -65,6 +65,7 @@ loongarch64_local_resume (unw_addr_space_t as, unw_cursor_t *cursor, void *arg)
         : "r" (uc->uc_mcontext.__gregs),
           "r" (sp),
           "r" (ra)
+        : "$t0", "$t1", "$t2", "memory"
```

Comment:
Beginning the 42th-line:
```suggestion
      /* Since there are no signals involved here we restore EH and non scratch
         registers only.  */
      register unsigned long src_t0 asm("$12") = (unsigned long)uc->uc_mcontext.__gregs;
      asm volatile (
        "ld.d $ra, $t0, 8\n"
        "ld.d $sp, $t0, 3*8\n"
        "ld.d $fp, $t0, 22*8\n"
        "ld.d $s0, $t0, 23*8\n"
        "ld.d $s1, $t0, 24*8\n"
        "ld.d $s2, $t0, 25*8\n"
        "ld.d $s3, $t0, 26*8\n"
        "ld.d $s4, $t0, 27*8\n"
        "ld.d $s5, $t0, 28*8\n"
        "ld.d $s6, $t0, 29*8\n"
        "ld.d $s7, $t0, 30*8\n"
        "ld.d $s8, $t0, 31*8\n"
        "jirl $r0, $ra, 0\n"
        :
        : "r" (src_t0)
      );
```

Reply:
@xen0n   Could you update this PR before merging?

Reply:
> @xen0n Could you update this PR before merging?

Sure. Just let me finish catching up with daily job and latest LoongArch developments after I returned from LPC...

Reply:
Updated, I went for an even simpler version that doesn't hand-allocate the `$t0`. Should be okay though I'm not sure if I should mark all of the saved registers clobbered. (I guess it's not necessary as the asm block is diverging so any of its side effects can't really affect the other code in the function.)

Reply:
> Updated, I went for an even simpler version that doesn't hand-allocate the `$t0`.

I suggested that ` register unsigned long src_t0 asm("$12") = (unsigned long)uc->uc_mcontext.__gregs;`
because maybe the `%0` is `ra` though it is rare.
I think that binding a fix register is more proper.

> Should be okay though I'm not sure if I should mark all of the saved registers clobbered. (I guess it's not necessary as the asm block is diverging so any of its side effects can't really affect the other code in the function.)

I think that is okay.

Reply:
> > Updated, I went for an even simpler version that doesn't hand-allocate the `$t0`.
> 
> I suggested that ` register unsigned long src_t0 asm("$12") = (unsigned long)uc->uc_mcontext.__gregs;` because maybe the `%0` is `ra` though it is rare. I think that binding a fix register is more proper.

Then how about specifying `$ra` as clobbered? I don't know if this is even possible but it should be more elegant than register allocation by hand IMO.

Reply:
> > > Updated, I went for an even simpler version that doesn't hand-allocate the `$t0`.
> > 
> > 
> > I suggested that ` register unsigned long src_t0 asm("$12") = (unsigned long)uc->uc_mcontext.__gregs;` because maybe the `%0` is `ra` though it is rare. I think that binding a fix register is more proper.
> 
> Then how about specifying `$ra` as clobbered?

There is no need to specify `$ra` as clobbered.

> I don't know if this is even possible but it should be more elegant than register allocation by hand IMO.

here is only restoring the caller-saved registers and the `if statement` is returned by inlined asm while not by high-level statement, using a temp register is ok.  I think that binding a fix register is more proper here.

Reply:
> > > Updated, I went for an even simpler version that doesn't hand-allocate the `$t0`.
> > 
> > 
> > I suggested that ` register unsigned long src_t0 asm("$12") = (unsigned long)uc->uc_mcontext.__gregs;` because maybe the `%0` is `ra` though it is rare. I think that binding a fix register is more proper.
> 
> Then how about specifying `$ra` as clobbered?

There is no need to specify `$ra` as clobbered.

> I don't know if this is even possible but it should be more elegant than register allocation by hand IMO.

here is only restoring the caller-saved registers and the `if statement` is returned by inlined asm while not by high-level statement, using a temp register is ok.  I think that binding a fix register is more proper here.

Reply:
I've fixed according to your suggestion as I have no more mental energy for this, and your suggested pattern looks reasonable too. Please check if the latest revision is acceptable for you.

Reply:
Thanks very much.

Maybe we can divide it into three parts：
(1) before the inline asm: here maybe the potential registers-conflict within the inline asm. so we add `register void *gregs asm("$t0") = uc->uc_mcontext.__gregs;`
(2) within the inline asm; what we want the asm code.
(3) after the inline asm. liking the `memory` is used for the after the inline asm. but this can't avoid the popotential registers-conflict before the inline asm.